corehq/apps/es/aggregations.py

```diff
@@ -0,0 +1,251 @@
+
+    @property
+    def buckets_dict(self):
+        return {b['key']: Bucket(b, self._aggregations) for b in self.normalized_buckets}
+
+    @property
+    def buckets_list(self):
+        return {Bucket(b, self._aggregations) for b in self.normalized_buckets}
+
+    @property
+    def raw_buckets(self):
+        return self.result['buckets']
+
+    @property
+    def normalized_buckets(self):
+        return self.raw_buckets
+
+    @property
+    def counts_by_bucket(self):
+        return {b['key']: b['doc_count'] for b in self.normalized_buckets}
+
+
+class Bucket(object):
+    def __init__(self, result, aggregations):
+        self.result = result
+        self.aggregations = aggregations
+
+    @property
+    def key(self):
+        return self.result['key']
+
+    @property
+    def doc_count(self):
+        return self.result['doc_count']
+
+    def __getattr__(self, attr):
+        sub_aggregation = filter(lambda a: a.name == attr, self.aggregations)[0]
+        if sub_aggregation:
+            return sub_aggregation.parse_result(self.result)
+
+
+class TermsAggregation(Aggregation):
+    """
+    Bucket aggregation that aggregates by field
+
+    :param name: aggregation name
+    :param field: name of the field to bucket on
+    """
+    type = "terms"
+    result_class = BucketResult
+
+    def __init__(self, name, field):
+        assert re.match(r'\w+$', name), \
+            "Names must be valid python variable names, was {}".format(name)
+        self.name = name
+        self.body = {
+            "field": field,
+        }
+
+
+class FilterResult(AggregationResult):
+    @property
+    def doc_count(self):
+        return self.result['doc_count']
+
+
+class FilterAggregation(Aggregation):
+    """
+    Bucket aggregation that creates a single bucket for the specified filter
+
+    :param name: aggregation name
+    :param filter: filter body
+    """
+    type = "filter"
+    result_class = FilterResult
+
+    def __init__(self, name, filter):
+        self.name = name
+        self.body = filter
+
+
+class FiltersAggregation(Aggregation):
+    """
+    Bucket aggregation that creates a bucket for each filter specified using
+    the filter name.
+
+    :param name: aggregation name
+    """
+    type = "filters"
+    result_class = BucketResult
+
+    def __init__(self, name, filters=None):
+        self.name = name
+        self.body = {"filters": (filters or {})}
+
+    def add_filter(self, name, filter):
+        """
+        :param name: filter name
+        :param filter: filter body
+        """
+        self.body["filters"][name] = filter
+        return self
+
+
+class AggregationRange(namedtuple('AggregationRange', 'start end key')):
+    def __new__(cls, start=None, end=None, key=None):
+        assert start or end, "At least one of 'from' or 'to' are required"
+        return super(AggregationRange, cls).__new__(cls, start, end, key)
```

Comment:
They still have `gt` `gte` etc for range filters. Just not for range aggregations